spyder/plugins/maininterpreter.py

```diff
@@ -158,8 +157,8 @@ def python_executable_switched(self, custom):
         if not is_text_string(cust_pyexec):
             cust_pyexec = to_text_string(cust_pyexec.toUtf8(), 'utf-8')
         if def_pyexec != cust_pyexec:
-            if custom:
-                self.warn_python_compatibility(cust_pyexec)
+            if custom and not self.warn_python_compatibility(cust_pyexec):
+                self.pyexec_edit.setText(def_pyexec)

```

Comment:
I think we should show the same warning as above before doing this, i.e.

    You selected an invalid Python interpreter for the console so the previous ...

Reply:
@ccordoba12   

`python_executable_changed` and `python_executable_switched` had very similar code, especially with the addition of this message.  Instead of duplicating the message, I instead changed it to call `python_executable_changed` from `python_executable_switched`.  I hope that's OK.